tutorials/transfer_learning/dataloader.jl

```diff
@@ -7,16 +7,46 @@ if isempty(readdir(PATH))
   error("Empty train folder - perhaps you need to download and extract the kaggle dataset.")
 end
 
+# Get all of the files with "dog" in the name
 const DOGS = filter(x -> occursin("dog", x), FILES)
+
+# Get all of the files with "cat" in the name
 const CATS = filter(x -> occursin("cat", x), FILES)
 
+# Takes in the number of requested images per batch ("n") and image size
+# Returns a 4D array with images and an array of labels
 function load_batch(n = 10, nsize = (224,224); path = PATH)
+  if ((batchsize % 2) != 0)
+      print("Batch size must be an even number")
+  end
+  # Sample N dog images and N cat images, shuffle, and then combine them into a batch
   imgs_paths = shuffle(vcat(sample(DOGS, Int(n/2)), sample(CATS, Int(n/2))))
```

Comment:
If `n == 1`, then this will produce an empty array for `img_paths`. It would be safer to do

```julia
ndogs = div(n, 2)
ncats = n - ndogs
img_paths = shuffle(vcat(sample(DOGS, ndogs), sample(CATS, ncats)))
```

You would also be able to remove the even restriction.